docs/content/contrib-code.md

```diff
@@ -0,0 +1,151 @@
+```
+
+### Developer Certificate Of Origin
+
+The OPA project requires that contributors sign off on changes submitted to OPA
+repositories.
+The [Developer Certificate of Origin (DCO)](https://developercertificate.org/)
+is a simple way to certify that you wrote or have the right to submit the code
+you are contributing to the project.
+
+The DCO is a standard requirement for Linux Foundation and CNCF projects.
+
+You sign-off by adding the following to your commit messages:
+
+    This is my commit message
+
+    Signed-off-by: Random J Developer <random@developer.example.org>
+
+Git has a `-s` command line option to do this automatically.
+
+    git commit -s -m 'This is my commit message'
+
+You can find the full text of the DCO here: https://developercertificate.org/
+
+### Code Review
+
+Before a Pull Request is merged, it will undergo code review from other members
+of the OPA community. In order to streamline the code review process, when
+amending your Pull Request in response to a review, do not squash your changes
+into relevant commits until it has been approved for merge. This allows the
+reviewer to see what changes are new and removes the need to wade through code
+that has not been modified to search for a small change.
+
+When adding temporary patches in response to review comments, consider
+formatting the message subject like one of the following:
+
+- `Fixup into commit <commit ID> (squash before merge)`
+- `Fixed changes requested by @username (squash before merge)`
+- `Amended <description> (squash before merge)`
+
+The purpose of these formats is to provide some context into the reason the
+temporary commit exists, and to label it as needing squashed before a merge
+is performed.
+
+It is worth noting that not all changes need be squashed before a merge is
+performed. Some changes made as a result of review stand well on their own,
+independent of other commits in the series. Such changes should be made into
+their own commit and added to the PR.
+
+If your Pull Request is small though, it is acceptable to squash changes during
+the review process. Use your judgement about what constitutes a small Pull
+Request.  If you aren't sure, send a message to the OPA slack or post a comment
+on the Pull Request.
+
+## Contribution process
+
+Small bug fixes (or other small improvements) can be submitted directly via a
+[Pull Request](https://github.com/open-policy-agent/opa/pulls) on GitHub.
+You can expect at least one of the OPA maintainers to respond quickly.
+
+Before submitting large changes, please open an issue on GitHub outlining:
+
+- The use case that your changes are applicable to.
+- Steps to reproduce the issue(s) if applicable.
+- Detailed description of what your changes would entail.
+- Alternative solutions or approaches if applicable.
+
+Use your judgement about what constitutes a large change. If you aren't sure,
+send a message to the [OPA slack](https://slack.openpolicyagent.org/) or submit
```

Comment:
We could be more specific now and call out the development channel?

Reply:
updated